jumpscale/packages/tfgrid_solutions/chats/discourse.py

```diff
@@ -0,0 +1,243 @@
+import math
+
+from jumpscale.loader import j
+from jumpscale.sals.chatflows.chatflows import GedisChatBot, chatflow_step, StopChatFlow
+import uuid
+from jumpscale.sals.reservation_chatflow import deployer
+import nacl
+import nacl.signing
+import random
+
+
+class Discourse(GedisChatBot):
+    FLIST_URL = "https://hub.grid.tf/omar0.3bot/omarelawady-discourse_all_in_one-latest.flist"
+
+    steps = [
+        "discourse_start",
+        "discourse_name",
+        "discourse_smtp_server",
+        "discourse_smtp_username",
+        "discourse_smtp_password",
+        "discourse_email",
+        "select_pool",
+        "discourse_network",
+        "overview",
+        "reservation",
+        "discourse_access",
+    ]
+
+    title = "Discourse"
+
+    @chatflow_step()
+    def discourse_start(self):
+        self.solution_id = uuid.uuid4().hex
+        self.user_form_data = dict()
+        self.query = dict()
+        self.user_form_data["chatflow"] = "discourse"
+        self.md_show("# This wizard will help you deploy discourse", md=True)
+        self.threebot_name = j.data.text.removesuffix(self.user_info()["username"], ".3bot")
+        self.solution_metadata = {}
+
+    @chatflow_step(title="Solution name")
+    def discourse_name(self):
+        self.solution_name = deployer.ask_name(self)
+
+    @chatflow_step(title="SMTP server")
```

Comment:
what about we merge smpt server, username, password, email into a form in one single page ? `form_ask`

Reply:
Looks much better. But email is not really a part of the SMTP configuration so I kept it in a separate step, do you think it's better to merge them?

Reply:
then all smpt stuff together in a form maybe ?
also for password we can use `secret_ask`

Reply:
done.